docs/proposals/proposal-001-trigger-and-deploy.md

```diff
@@ -0,0 +1,248 @@
+
+### Linked Docs
+
+- **Slack Discussion Thread** [Link](https://cloud-native.slack.com/archives/C060EDHN431/p1712765271470189)
+
+## Proposal
+
+We will watch for new releases of the project by subscribing to the Atom feed
+of releases that GitHub publish e.g. https://github.com/falcosecurity/falco/releases.atom
+
+Our automation will call the GitHub REST API to trigger the pipeline.
+
+The maintainers of the CNCF projects will also be able to call this webhook
+using a fine grained access token we will provide.
+
+This can be used to trigger the pipeline ad-hoc during development and can be
+added to their CI/CD pipeline if additional trigger points are required.
+
+### User Stories
+
+#### Project maintainer creates new release to be measured
+
+Our automation detects a new release was published and triggers the pipeline.
+
+#### Project maintainer deploys their project so it can be measured
+
+Participating CNCF projects will deploy their project using a gitops approach
+with flux.
+
+#### Project maintainer triggers pipeline to test a new benchmark
+
+Calling the webhook will trigger the pipeline allowing the changes to be tested.
+
+### Risks and Mitigations
+
+Multiple deployments will produce inaccurate results as we can only accurately
+measure a single project per node. We can set [concurrency](https://docs.github.com/en/actions/using-jobs/using-concurrency)
+in the workflow to ensure only a single instance runs at a time.
+
+Uninstall at end of pipeline fails. We can wait till all flux finalizers are
+removed. In future we could create nodes on demand and delete on completion.
+
+## Design Details
+
+![Pipeline Design](./files/trigger-deploy-pipeline.svg)
+
+### Subscribing to Releases
+
+A YAML file of CNCF projects and any sub components will be stored in the
+tooling repo e.g.
+
+```yaml
+# projects.yaml
+projects:
+  - name: falco
+    sub_components:
+      - ebpf
+      - modern-ebpf
+      - kmod
+```
+
+A scheduled GitHub Action will run every 30 minutes and check the Atom feed of
+each project for new releases. To manage the state a GitHub [repository variable](https://docs.github.com/en/actions/learn-github-actions/variables)
+per CNCF project is used to store the latest release version.
+
+If a new release is detected the action will trigger the pipeline for the new
+release and update the variable with the new version. This is to ensure each
+release is only triggered once.
+
+If sub components are specified then the pipeline will be triggered once per
+sub component.
+
+### Trigger
+
+The green reviews pipeline will be triggered by sending a [workflow_dispatch](https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28#create-a-workflow-dispatch-event)
+event via the GitHub REST API.
+
+Inputs are
+
+- `cncf_project`: **required** Project to be deployed e.g. `falco`
+- `cncf_project_sub`: **optional** Subcomponent if project has multiple variants
```

Comment:
Is this a list? Shall we add it in the docs? Would it be helpful to have an example with multiple project_sub?

Reply:
@ADiTuri Thank you that needed clarifying.

Its an optional list. Falco have 3 drivers they wish to use but so far we only deploy a single driver. 
I added this example to make it clearer. Does that help?